client/db/src/lib.rs

```diff
@@ -1257,13 +1261,23 @@ impl<Block: BlockT> Backend<Block> {
 
 			let finalized = if operation.commit_state {
 				let mut changeset: sc_state_db::ChangeSet<Vec<u8>> = sc_state_db::ChangeSet::default();
+				let mut ops: u64 = 0;
+				let mut bytes: u64 = 0;
 				for (key, (val, rc)) in operation.db_updates.drain() {
 					if rc > 0 {
+						ops += 1;
```

Comment:
Field db_updates are calculated trie nodes delta that are going to be written in rocksdb, so I think we should change the name, because it gives the wrong idea that the collected read stats (key value) are comparable with those writes (trie nodes).
We can also get write stats that correspond to our read, that would be field 'storage_updates' and 'child_storage_updates', IIRC those are send to storage cache for update and are complete in term of delta.
But we could also hook some 'write_cache' stat in state-machine overlay-db to compare what are actual post block modif ('storage_updates' fields) against what update the runtime did call. That way, we get an idea of the proportion of updates that are done on a same key. (at https://github.com/paritytech/substrate/blob/65ad8e9632f1a627fc9fb386dc5b60a54741d7a0/primitives/state-machine/src/overlayed_changes.rs#L235 and the child one)